src/index.ts

```diff
@@ -248,21 +251,23 @@ class Compiler {
   }
 
   private validateEnumMembers(rule: JSONSchema.Schema): EnumType {
-    if (!rule.type) rule.type = 'string'
+    if (rule.type === undefined) rule.type = 'string'
```

Comment:
if `"type" : "boolean"`,  it will be defined, so `rule.type` won't default to `string`.